### DISCUSSION TITLE
Autoanswer with automatically play Audiofile

### DISCUSSION THREAD

**Discussion by Evope** [baresip/baresip#1873] [Q&A] [answered]
Hello everyone.
I have in mind to use baresip for a VoIP quality measurement. Therefore the software should register a SIP account and if I make a call to that account it should answere und play a defined audiofile.

The installation on a raspbarry pi was successful and I also registered my SIP account. With `answermode=auto` it also answeres my calls.
Now I thought that `sip_autoanswer_aufile` in the Config-file activates and set the audiofile that is played after answering. But here comes my problem: it does not.

Did I missed something? Do I have to activate it otherwise? Does baresip even have the possibility to play a wave-file after auto-answering a SIP call?

Thanks in advance for your help <3

**Comment by sreimers** [comment]
See: https://github.com/baresip/baresip/wiki/Accounts#sip_autoanswer

**Reply by Evope** [reply]
Thanks for you help Sebastian!
But, as I said, it already answeres my calls. It just don't play the defined wave-file.
Maybe I have to take a closer look to that alsa-module, because I do not even hear that autoanswer-beep-sound.

**Reply by sreimers** [reply]
sorry maybe not detailed enough, you can't use `sip_autoanswer_aufile` with `answermode=auto` together, as mentioned in the docs:
```
Note: If answermode=auto then this setting is ignored.
```
Try `sip_autoanswer=yes` (instead of answermode) and `sip_autoanswer_aufile`

**Reply by Evope** [reply]
Ok, I see that `sip_autoanswer=yes` and `sip_autoanswer_aufile` have the same context.
But for me, `sip_autoanswer=yes` is not working in contrast to `answermode=auto` which works.
I attached two videos with both methods so you can compare.

`<sip:siptester@192.168.178.1:5060;transport=tcp>;auth_pass=Siptester-2022;sip_autoanswer=yes`
https://user-images.githubusercontent.com/106373272/170974227-45a3ece0-30c1-4dd7-9443-1af77ac7e780.mp4
(I pressed 'a' to accept incomming call)

`<sip:siptester@192.168.178.1:5060;transport=tcp>;auth_pass=Siptester-2022;answermode=auto`
https://user-images.githubusercontent.com/106373272/170974332-c7ba74d4-199c-4f9f-bcd9-32f5865b0f1f.mp4

My config file:
```
# Audio
audio_source		alsa,file:///home/pi/test.wav

# Menu
sip_autoanswer_aufile	test.wav
```

Sounds like I also have a problem with audiotransmitting. You can just hear an annoying noise. :(

**Comment by juha-h** [comment]
I have used in config file:
```
audio_source		gst,file:///tmp/digits.wav
```
and in account:
```
;answermode=auto
```

**Reply by Evope** [reply]
Thanks, I will try that audio_source parameter!

**Reply by cspiel1** [reply]
Like @sreimers already tried to explain `answermode=auto` has not much to do with the SIP auto answer header.
https://github.com/baresip/baresip/wiki/Accounts#sip_autoanswer

The latter has to be configured on the callee side. And on the caller side the `menu` command `setansval` has to be used to add the auto answer SIP header.

**Reply by Evope** [reply]
> The latter has to be configured on the callee side. And on the caller side the `menu` command `setansval` has to be used to add the auto answer SIP header.

Oh well, I think I got it now. `sip_autoanswer` is kind of a feature in the SIP protocol. So when the caller don't send an autoanswer request in the INVITE, the baresip UA will not automatically answer. 

Never heard that before so please excuse the disorder.

**Comment by cspiel1** [comment]
This makes no sense:
```
audio_source		alsa,file:///home/pi/test.wav
```

Module `alsa` always uses the Microphone as source. Not a WAV file.

**Reply by Evope** [reply]
Yeah that was my fault, it was a kind of trial an error. Sorry for that. It was just the first configuration I heard anything with.

**Comment by sreimers** [comment, marked as answer]
You can use these account params (if you don't want to use `sip_autoanswer` but please don't mix these params/configs):

`;audio_source=aufile,/tmp/file.wav;answermode=auto`

and maybe use aubridge module as a dummy device (if needed):

~/.baresip/config:
```
audio_player           aubridge,pseudo0
audio_source          aubridge,pseudo0
audio_alert             aubridge,pseudo0
```

**Reply by Evope** [reply]
It works, thank you.
That hint with aubridge module was great, thanks for that!
